acceptance/tests/allow_arbitrary_node_name_fact_for_agent.rb

```diff
@@ -25,6 +25,7 @@
   teardown do
     modify_tk_config(master, options['puppetserver-config'], {'jruby-puppet' => {'use-legacy-auth-conf' => true}})
     on master, 'cp /etc/puppetlabs/puppetserver/conf.d/auth.bak /etc/puppetlabs/puppetserver/conf.d/auth.conf'
+    on master, 'service puppetserver reload'
```

Comment:
it used to be (and may still be?) that PE tests reused some of our tests, in which case the service may be named `pe-puppetserver`. To account for that difference, we try to use https://github.com/puppetlabs/puppet/blob/fc8dd4771cf736e44b20d4c017d79d574d223b70/acceptance/tests/language/exported_resources.rb#L18.

One unfortunate thing about this is we'll end up restarting the service twice. Once in the `with_puppet_running_on`](it used to be (and may still be?) that PE tests reused some of our tests, in which case the service may be named `pe-puppetserver`. To account for that difference, we try to use https://github.com/puppetlabs/puppet/blob/fc8dd4771cf736e44b20d4c017d79d574d223b70/acceptance/tests/language/exported_resources.rb#L18.

One unfortunate thing about this is we'll end up restarting the service twice. Once in the `with_puppet_running_on`](it used to be (and may still be?) that PE tests reused some of our tests, in which case the service may be named `pe-puppetserver`. To account for that difference, we try to use https://github.com/puppetlabs/puppet/blob/fc8dd4771cf736e44b20d4c017d79d574d223b70/acceptance/tests/language/exported_resources.rb#L18.

One unfortunate thing is we'll end up restarting the puppetserver service twice. Once in the [`with_puppet_running_on`](https://github.com/voxpupuli/beaker-puppet/blob/376a735afee0735dc7e7547310a060a738d6f5da/lib/beaker-puppet/helpers/puppet_helpers.rb#L204) and again in the `teardown`, which can slow the test down... It's not a blocking comment, just curious if there's a way to only restart once?

Reply:
I don't think so... from what I know the service needs to be reloaded to pick up `auth.conf` changes. However a reload is somewhat faster than a full restart:

```
[root@rid-metronome ~]# time service puppetserver reload
Redirecting to /bin/systemctl reload  puppetserver.service

real	0m5.222s
user	0m0.008s
sys	0m0.016s
[root@rid-metronome ~]# time service puppetserver restart
Redirecting to /bin/systemctl restart  puppetserver.service

real	0m18.606s
user	0m0.009s
sys	0m0.022s
```